1_prep/src/munge_netCDFs.R

```diff
@@ -1,29 +1,155 @@
-#' the feather file output of the split netCDF data. The first `%s` is
-#' used as the placeholder for the `gcm_name`, the second is for the `gcm_date`,
-#' and the third is for the `cell_no`.
-#' @return names of cell/gcm/time-period specific feather files
-munge_nc_files <- function(gcm_nc, gcm_name, cell_no, gcm_date, outfile_template) {
-  # open netCDF file (gcm-specific) and pull data for that cell, for that time period
-  # TODO - actually pull time data for only one time period - pulling all for now
-  nc <- nc_open(gcm_nc)
-  time_dim <- ncvar_get(nc, "time")
-  cell_dim <- ncvar_get(nc, "gridcells")
-  data <- purrr::map_df(nc$var, function(var) {
-    ncvar_get(nc, var, start = c(1,cell_no), count = c(-1,1))
-  }) %>% mutate(time = nc.get.time.series(f = nc,time.dim.name = "time"),.before=1)
+#' the csv file output of the split netCDF data. The first `%s` is
+#' used as the placeholder for the `gcm_name`, the second is for the time period,
+#' and the third is for the data_cell_no
+#' @return names of gcm/time-period/cell-specific csv files
+munge_nc_files <- function(gcm_nc, gcm_name, cell_nos, gcm_time_periods, burn_in, burn_out, outfile_template) {
+  # Open the netCDF file for the current GCM
+  nc <- ncdfgeom::read_timeseries_dsg(gcm_nc)
+  
+  # Pull the vectors of dates
+  nc_dates <- lubridate::as_date(nc$time)
+  
+  # Generate a list of time-period specific and cell-specific files
+  # For each cell...
+  time_period_cell_files <- purrr::map(cell_nos, function(cell_no) {
+    # For each variable stored in the netCDF...
+    cell_files <- purrr::map2_dfc(nc$data_frames, names(nc$data_frames), function(var_df, var_name) {
+      # Pull the column pertaining to the current cell and rename it to the variable name
+      var_cell_long <- var_df %>%
+        select(as.character(cell_no)) %>%
+        rename(!!var_name := as.character(cell_no))
+    }) %>%
+      # Once all variables have been compiled for each cell,
+      # Add a column for time and determine the time period for each row
+      mutate(time = nc_dates,
+             time_period = case_when(
+               time <= as.Date("2010-01-01") ~ gcm_time_periods[1],
+               time <= as.Date("2070-01-01") ~ gcm_time_periods[2],
+               TRUE ~ gcm_time_periods[3])
+             ) %>%
+      # Reorder the columns to match the standard order
+      select(time_period, time, Shortwave, Longwave, AirTemp, RelHum, WindSpeed, Rain, Snow) %>%
+      # Group by time period
+      group_by(time_period) %>%
+      # For each time period...
+      group_map(~ {
+        time_period <- .y$time_period
+        # add the specified number of burn_in and burn_out days
+        meteo_w_burn_in_out <- add_burn_in_out_to_meteo(.x)
+        # then write the data for the current cell to a csv file
+        outfile <- sprintf(outfile_template, gcm_name, time_period, cell_no)
+        fwrite(setDT(meteo_w_burn_in_out), outfile)
```

Comment:
I thought we had decided `setDT()` was not needed to use `fwrite()`?

Reply:
Oh I just pulled that straight from the example you shared in our chat. I'll test it without `setDT()`